packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -66,14 +66,18 @@ export function esbuildDepPlugin(
   return {
     name: 'vite:dep-pre-bundle',
     setup(build) {
+      const nonJsFileRE = new RegExp(
+        `\\.(` + externalTypes.join('|') + `)(\\?.*)?$`
+      )
       // externalize assets and commonly known non-js file types
       build.onResolve(
         {
-          filter: new RegExp(`\\.(` + externalTypes.join('|') + `)(\\?.*)?$`)
+          filter: nonJsFileRE
```

Comment:
Is this refactoring necessary?

Reply:
It is reused in line `80`
So IMO it is a good solution so the pattern don't need to be compiled two times

Or do you addressing something different?

Reply:
nvm, my bad 😅